lib/src/Base/Stat/LatentVariableModel.cxx

```diff
@@ -0,0 +1,299 @@
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/LatentVariableModel.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/Exception.hxx"
+#include "openturns/SpecFunc.hxx"
+#include "openturns/Polygon.hxx"
+#include "openturns/Text.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(LatentVariableModel)
+
+static const Factory<LatentVariableModel> Factory_LatentVariableModel;
+
+/** Parameters constructor */
+LatentVariableModel::LatentVariableModel(const UnsignedInteger nLevels,
+										 const UnsignedInteger latentDim)
+  : CovarianceModelImplementation(Point(1, 1.0), Point(1, 1.0))
+  , latentDim_(latentDim)
+  , nLevels_(nLevels)
+  , latentCovarianceMatrix_(nLevels_)
+  , latentCovarianceModel_(latentDim_)
+{
+  if (latentDim_ < 1) throw InvalidArgumentException(HERE) << "Error: the dimension of the latent space must be >= 1";
+  if (nLevels_ < 2) throw InvalidArgumentException(HERE) << "Error: the number of discrete levels must be >= 2";
+  activeLatentCoordinateDim_ = 1 + latentDim_ * (nLevels_ - 2);
+  activeLatentVariables_ = Point(activeLatentCoordinateDim_, 0.0);
+  fullLatentVariables_ = Sample(nLevels_, latentDim_);
+  Indices activeParameter = Indices(inputDimension_ + outputDimension_ + activeLatentCoordinateDim_);
+  activeParameter.fill();
+  setActiveParameter(activeParameter);
+  updateLatentCovarianceMatrix();
+}
+
+/* Virtual constructor */
+LatentVariableModel * LatentVariableModel::clone() const
+{
+  return new LatentVariableModel(*this);
+}
+
+/* Computation of the covariance  function */
+Scalar LatentVariableModel::computeAsScalar(const Scalar z1, const Scalar z2) const
+{
+  /*checks z1 and z2 being among the known levels*/
+  const bool isLevelz1 = (z1 >= 0 && z1 < nLevels_ && (z1 - floor(z1)) == 0);
+  const bool isLevelz2 = (z2 >= 0 && z2 < nLevels_ && (z2 - floor(z2)) == 0);
+
+  if (!isLevelz1 || !isLevelz2) throw InvalidArgumentException(HERE) << "Error: the input discrete variables values: " << z1 
+  << ", and/or " << z2 << " are not among the known levels. They should both present integer values between 0 and l-1.";
+
+  Scalar cov = latentCovarianceMatrix_(z1, z2);
+
+  return cov;
+}
+
+Scalar LatentVariableModel::computeAsScalar(const Point & z1, const Point & z2) const
+{
+  return computeAsScalar(z1[0], z2[0]);
+}
+
+
+Scalar LatentVariableModel::computeAsScalar(const Collection<Scalar>::const_iterator & z1_begin,
+    const Collection<Scalar>::const_iterator & z2_begin) const
+{
+
+  Collection<Scalar>::const_iterator z1_it = z1_begin;
+  Collection<Scalar>::const_iterator z2_it = z2_begin;
+
+  return computeAsScalar(*z1_it, *z2_it);
+}
+
+
```

Comment:
At this point, a one-line comment of the method is mandatory, as for all C++ methods in the cxx.
```suggestion
/* Set the vector of parameters */
```
Please add this for all methods below: some have this comment, but many do not.